app/models/npq_application.rb

```diff
@@ -115,4 +135,10 @@ def push_enrollment_to_big_query
       NPQ::StreamBigQueryEnrollmentJob.perform_later(npq_application_id: id)
     end
   end
+
+  def validate_funding_eligiblity_status_code_change
+    if declared_as_payable? && eligible_for_funding == false
+      errors.add(:funding_eligiblity_status_code)
```

Comment:
is there an error message we'd like to add here, also should we add this to base as it involves two fields?

Reply:
For now the way it works is that we don't allow to navigate to the page where this validation might be triggered, its more as a failsafe. But its a good idea and I will implement as suggested.